checker/src/org/checkerframework/checker/lock/wholefiles/LockVisitor.java

```diff
@@ -0,0 +1,580 @@
+                        TreeUtils.elementFromUse(tree),
+                        tree.getKind() == Tree.Kind.METHOD_INVOCATION,
+                          generatePreconditionsBasedOnGuards(valueType, false));
+
+                skipSubtypeCheck = true;
+            }
+        }
+
+        if (!skipSubtypeCheck) {
+            super.commonAssignmentCheck(varType, valueType, valueTree, errorKey, isLocalVariableAssignement);
+        }
+    }
+
+    @Override
+    public Void visitMemberSelect(MemberSelectTree node, Void p) {
+        checkAccessOfExpression(node);
+
+        return super.visitMemberSelect(node, p);
+    }
+
+    private void reportFailure(/*@CompilerMessageKey*/ String messageKey,
+            MethodTree overriderTree,
+            AnnotatedDeclaredType enclosingType,
+            AnnotatedExecutableType overridden,
+            AnnotatedDeclaredType overriddenType,
+            List<String> overriderLocks,
+            List<String> overriddenLocks
+            ) {
+        // Get the type of the overriding method.
+        AnnotatedExecutableType overrider =
+            atypeFactory.getAnnotatedType(overriderTree);
+
+        if (overrider.getTypeVariables().isEmpty()
+                && !overridden.getTypeVariables().isEmpty()) {
+            overridden = overridden.getErased();
+        }
+        String overriderMeth = overrider.toString();
+        String overriderTyp = enclosingType.getUnderlyingType().asElement().toString();
+        String overriddenMeth = overridden.toString();
+        String overriddenTyp = overriddenType.getUnderlyingType().asElement().toString();
+
+        if (overriderLocks == null || overriddenLocks == null) {
+            checker.report(Result.failure(messageKey,
+                    overriderMeth, overriderTyp,
+                    overriddenMeth, overriddenTyp), overriderTree);
+        }
+        else {
+            checker.report(Result.failure(messageKey,
+                    overriderMeth, overriderTyp,
+                    overriddenMeth, overriddenTyp,
+                    overriderLocks, overriddenLocks), overriderTree);
+        }
+    }
+
+    // Ensures that subclass methods require a subset of the locks that a parent class method requires
+    // Additionally ensures that subclass methods use @HoldingOnEntry (and not @Holding) if the parent
+    // class method uses @HoldingOnEntry.
+    @Override
+    protected boolean checkOverride(MethodTree overriderTree,
+            AnnotatedDeclaredType enclosingType,
+            AnnotatedExecutableType overridden,
+            AnnotatedDeclaredType overriddenType,
+            Void p) {
+
+        // Check @Holding overrides
+
+        List<String> overriderLocks = atypeFactory.methodHolding(TreeUtils.elementFromDeclaration(overriderTree));
+        List<String> overriddenLocks = atypeFactory.methodHolding(overridden.getElement());
+
+        boolean isValid = true;
+
+        if (!overriderLocks.isEmpty()) {
```

Comment:
This isn't need.  (And it was confusing me as to why the empty set was special cased.)